src/beast/beast/unit_test/suite.h

```diff
@@ -346,11 +346,13 @@ bool
 suite::expect (Condition shouldBeTrue,
     String const& reason)
 {
+    bool const b =
+        static_cast<bool>(shouldBeTrue);
     if (shouldBeTrue)
```

Comment:
Feels like you should be testing `b` in the `if`, rather than `shouldBeTrue`.  The results should be the same, but you're losing part of the efficiency of pre-computing `b`.  You'll also guarantee consistency between the `pass\fail` result and the returned `bool`.